src/code/z_map_disp.c

```diff
@@ -1,109 +1,1881 @@
+    s32 posY;
+    s32 texOffsetX;
+    s32 texOffsetY;
+    s32 texWidth;
+    s32 texHeight;
+    s32 scale;
+    f32 scaleFrac;
+
+    mapDataRoom = &sMapDisp.mapDataScene->rooms[sMapDisp.curRoom];
+    if (mapDataRoom->mapId == 0xFFFF) {
+        return;
+    }
+
+    MapDisp_GetMapOffset(mapDataRoom, &texOffsetX, &texOffsetY);
+    MapDisp_GetMapTexDim(mapDataRoom, &texWidth, &texHeight);
+    scale = sMapDisp.mapDataScene->scale;
+
+    if (sMapDisp.mapDataScene->scale == 0) {
+        scale = 20;
+    } else if (sMapDisp.mapDataScene->scale == -1) {
+        s32 scaleTemp;
+
+        MapDisp_GetMapScale(mapDataRoom, &scaleTemp);
+        scale = scaleTemp;
+    }
+
+    scaleFrac = 1.0f / scale;
+    if (!MapDisp_IsDataRotated(play)) {
+        posX = (s32)((x - (f32)mapDataRoom->centerX) * scaleFrac) + sMapDisp.minimapBaseX +
+               (sMapDisp.minimapCurX - sMapDisp.minimapBaseX) + texOffsetX;
+        posY = (s32)((z - (f32)mapDataRoom->centerZ) * scaleFrac) + sMapDisp.minimapBaseY +
+               (sMapDisp.minimapCurY - sMapDisp.minimapBaseY) + texOffsetY;
+    } else {
+        posX = -(s32)((x - (f32)mapDataRoom->centerX) * scaleFrac) + sMapDisp.minimapBaseX +
+               (sMapDisp.minimapCurX - sMapDisp.minimapBaseX) + texOffsetX;
+        posY = -(s32)((z - (f32)mapDataRoom->centerZ) * scaleFrac) + sMapDisp.minimapBaseY +
+               (sMapDisp.minimapCurY - sMapDisp.minimapBaseY) + texOffsetY;
+    }
+
+    if ((posX > 0) && (posX < 0x3FF) && (posY > 0) && (posY < 0x3FF)) {
+        OPEN_DISPS(play->state.gfxCtx);
+
+        Gfx_SetupDL42_Overlay(play->state.gfxCtx);
+        gSPMatrix(OVERLAY_DISP++, &gIdentityMtx, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gDPSetCombineLERP(OVERLAY_DISP++, PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE, 0,
+                          PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE, 0);
+        gDPSetEnvColor(OVERLAY_DISP++, 0, 0, 0, 255);
+        gDPSetCombineMode(OVERLAY_DISP++, G_CC_PRIMITIVE, G_CC_PRIMITIVE);
+        gDPSetRenderMode(OVERLAY_DISP++, G_RM_AA_DEC_LINE, G_RM_NOOP2);
+        Matrix_Translate(posX - 160.0f, 120.0f - posY, 0.0f, MTXMODE_NEW);
+        Matrix_RotateXFApply(-1.6f);
+        if (MapDisp_IsDataRotated(play)) {
+            rot += 0x7FFF;
+        }
+        Matrix_RotateYF(rot / 10.0f, MTXMODE_APPLY);
+        Matrix_Scale(0.4f, 0.4f, 0.4f, MTXMODE_APPLY);
+        gSPMatrix(OVERLAY_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gDPSetPrimColor(OVERLAY_DISP++, 0, 255, 200, 0, 0, play->interfaceCtx.minimapAlpha);
+        gSPDisplayList(OVERLAY_DISP++, gCompassArrowDL);
+
+        CLOSE_DISPS(play->state.gfxCtx);
+    }
+}
+
+s32 MapDisp_IsLocationRomaniRanchCutscene(PlayState* play) {
+    if ((gSaveContext.save.entrance == ENTRANCE(ROMANI_RANCH, 0)) && (Cutscene_GetSceneLayer(play) != 0)) {
+        return true;
+    }
+    return false;
+}
```

Comment:
Just struggling to think of what to name the check in `MapDisp_CanDrawDoors` and here. Strictly speaking `Cutscene_GetSceneLayer(play) != 0` covers alt layers that are both playable and not playable.

Reply:
This is just the opposite of `MapDisp_CanDrawDoors`, so if the other is named `MapDisp_CanDrawDoors` this should probably `MapDisp_CannotDrawDoors`. Those actually don't seem bad as it seems to be the purpose of the functions.

Reply:
Actually looking at use for this perhaps something like `MapDisp_IsLocationRomaniRanchAltScene`, since this one doesn't seem to be related to drawing doors (while the other one definitely is).